File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Rectangle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Rectangle implements Shape {
    public double height;
    public double width;
    
    public Rectangle(double height, double width){
        this.height = height;
        this.width = width;
    }

    public double getArea() {
        return height * width;
    }
    public int getShape()
    {
        return 1;
    }

    public double[] getSize()
    {
        return new double[]{height,width};
    }

}

File: typescript/src/shapes/rectangle.ts
import { Shape,ShapeType } from "./shape.js";

function Rectangle(width: number, height: number): Shape {

    return {
        
        computeArea(): number {
            return width * height
        },
        getShape(): ShapeType {
            return 'circle'
          },
        getSize(): number[] {
            return [width,height]
          },
        };
}

export { Rectangle }
